client/src/components/profileInfoEdit/ProfileInfoEdit.js

```diff
@@ -18,31 +18,70 @@ import { updateTeam } from '../../actions/team';
 const ProfileInfoEdit = () => {
   const dispatch = useDispatch();
 
-  const { teamId, teamName, orgName, email } = useSelector(
+  const { teamId, teamName, orgName, email,password, confirmedPassword } = useSelector(
     (state) => state.team
   );
 
-  const [profileData, setInputs] = useState({ teamName, orgName, email });
+  const [profileData, setInputs] = useState({ teamName, orgName, email,password ,confirmedPassword});
 
   useEffect(() => {
-    setInputs({ teamName, orgName, email });
-  }, [email, orgName, teamName]);
+    setInputs({ teamName, orgName, email,password,confirmedPassword  });
+  }, [email, orgName, teamName,password,confirmedPassword]);
 
   const updateInputs = (form) => {
     const { name, value } = form.target;
-    setInputs({ ...profileData, [name]: value });
+    if (name !== "email" ){
+      setInputs({ ...profileData, [name]: value });
+    }
   };
+  const checkPassword = function(){
 
-  const [validated, setValidated] = useState(false);
-  const handleUpdate = (event) => {
-    const form = event.currentTarget;
-    event.preventDefault();
-    if (form.checkValidity() === false) {
-      event.stopPropagation();
-    } else {
-      dispatch(updateTeam(teamId, profileData));
+    if (! ({...profileData}.password === {...profileData}.confirmedPassword)){
+      return false;
     }
-    setValidated(true);
+    var matches = {...profileData}.password.match(
+        /^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$/);
+    if (matches === null){
+      console.log("false");
+      return false;
+    }
+    return true;
+
+  };
+
+  const [validated, setValidated] = useState(false);
+  const handleUpdate = async (event) => {
+      const form = event.currentTarget;
+      event.preventDefault();
+      if (form.checkValidity() === false) {
+          event.stopPropagation();
+      } else {
+          var newdata;
+          if (({...profileData}.password || {...profileData}.confirmedPassword ) && ({...profileData}.password !== "" || {...profileData}.confirmedPassword !== "")) {
+              if (!checkPassword()) {
+                    setValidated(false);
+                    alert("please enter a password at least 8 chars long, using only numbers, letters and characters");
+                    return;
+              } else {
+                  newdata = {
+                      "teamName": {...profileData}.teamName,
+                      "orgName": {...profileData}.orgName,
+                      "email": {...profileData}.email,
+                      "password": {...profileData}.password
+                  }
+                  dispatch(updateTeam(teamId, newdata));
+                  alert("password changed");
```

Comment:
Could we perhaps use a success toast here for consistency with the rest of the frontend?

Reply:
how do i use the success and error toasts?

Reply:
I think you can use errorActionGlobalCreator and successMessageCreator to make the toast. (Example: https://github.com/Researchify/Researchify/pull/174/commits/3d730e22af02dc57390ddf60e982755555695a36)

Reply:
I think you can use errorActionGlobalCreator and successMessageCreator to do the toast. (Example: https://github.com/Researchify/Researchify/pull/174/commits/3d730e22af02dc57390ddf60e982755555695a36)